### ISSUE TITLE
auth: complete user metadata public model

### ISSUE THREAD

**Issue by syl987** [firebase/firebase-js-sdk#7314]
### [REQUIRED] Describe your environment

  * Firebase SDK version: @angular/fire@7.5.0
  * Firebase Product: auth

Hi there,

sorry guys, it was unintentional to create a PR on the official repository. Looks like it was auto-proposed by Github :(

Link: https://github.com/firebase/firebase-js-sdk/pull/7313

I am using Firebase Auth module and displaying user's registration and last login dates. However, it seems that the timestamp properties are missing in the public model. Only the locale-specific properties are listed.

Kind Regards,
Igor

#### Steps to reproduce:

On runtime, check properties of `User.metadata`, 4 are listed but only 2 of them are included in the public model.

![console user-metadata](https://github.com/firebase/firebase-js-sdk/assets/53560066/f1e10c62-c68c-425c-8bac-8094f5cb32e3)

#### Relevant Code:

https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth/src/model/public_types.ts#L1094
https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth/src/core/user/user_metadata.ts#L22

**Comment by prameshj** [comment]
Is the goal to get the timestamps in UTC milliseconds instead of a date string? You can also get this from the idToken using the fields "auth_time" and "iat". Would that work?

**Comment by syl987** [comment]
Hi there,
no problem on my side, I am fine because this can be easily addressed by casting / custom typing. My use case was a simple mapping to a smaller, serializable local user object, so I went with what seemed simple.

My motivation to create this issue was to save other's time because I needed to do a bit of research on this once I noticed differences between the object and its typing.

I don't have any additional info on this outside of the accidentally created PR. Feel free to keep it or close it, as you prefer.

Have a good day!
